File: c/numericConverter.c
#include <stdio.h>
#include <math.h>
#include <ctype.h>
#include <string.h>


int len(int n);
int convert(int numToConvert, int base);
int convertFromString(char numToConvert[], int base);
void toUppercase(char str[]);

int main(void) {
    printf("From binary: %d\n", convert(101, 2));
    printf("From hexadecimal: %d\n", convertFromString("AF", 16));
}

int convert(int numToConvert, int base) {
    int sum = 0;

    for (int i = 0, length = len(numToConvert); i < length; i++) {
        int digit = numToConvert % 10;
        numToConvert /= 10;
        sum += digit * pow(base, i);
    }
    return sum;
}

int convertFromString(char numToConvert[], int base) {
    char num[strlen(numToConvert) + 1];
    strcpy(num, numToConvert);

    int sum = 0;

    for (int i = 0, length = strlen(num); i < length; i++) {
        int digit = num[length - i - 1];
        if (isalpha(digit)) {
            int toDecimal = 10 + digit - 'A';
            sum += toDecimal * pow(base, i);
        }
        else if (isdigit(digit)) {
            int toDecimal = digit - '0';
            sum += toDecimal * pow(base, i);
        }
    }
    return sum;
}

int len(int n) {
    int count = 0;

    if (n == 0)
        return 1;

    while (n != 0) {
        n /= 10;
        count++;
    }

    return count;
}

void toUppercase(char str[]) {
    for (int i = 0; str[i]; i++)
        str[i] = toupper(str[i]);
}

File: java/NumericConverter.java
public class NumericConverter {

    // Numeric Converter, be sure to insert the right numeric base to precise conversion!

    public static void main(String... args) {
        int num = 101;
        String hexadecimal = "AF";
        System.out.println(convert(num, 2));
        System.out.println(convert(hexadecimal, 16));
    }

    /*
     *   Handles conversion with a long value as paramater
     *   But only supports until decimal base
     *   If the base is higher than 10, the result will not be right
     */
    public static long convert(long numToConvert, int base) {
        String num = Long.toString(numToConvert);
        long sum = 0;
        
        for (int i = num.length() - 1; i >= 0; i--) {
            int digit = Integer.parseInt(String.valueOf(num.charAt(i)));
            sum += digit * Math.pow(base, num.length() - i - 1);
        }
        return sum;
    }

    // The same applies to this, but with an int value as paramater
    public static int convert(int numToConvert, int base) {
        String num = Long.toString(numToConvert);
        int sum = 0;
        
        for (int i = num.length() - 1; i >= 0; i--) {
            int digit = Integer.parseInt(String.valueOf(num.charAt(i)));
            sum += digit * Math.pow(base, num.length() - i - 1);
        }
        return sum;
    }

    /*
     * Handles conversion with a String number parameter, also supports
     * alphanumeric values, hexadecimal, for example.
     * Supports until base 36, if base is higher than 36 it will not work as expected
     */
    public static long convert(String numToConvert, int base) {
        if (numToConvert == null)
            throw new IllegalArgumentException("num argument cannot be null");

        String num = numToConvert.toUpperCase();
        int sum = 0;
        
        for (int i = 0, length = num.length(); i < length; i++) {
            int digit = num.charAt(length - i - 1);
            if (Character.isAlphabetic(digit)) {
                sum += (10 + digit - 'A') * Math.pow(base, i);
            }
            else if (Character.isDigit(digit)) {
                digit = Integer.parseInt(String.valueOf(num.charAt(i)));
                sum += digit * Math.pow(base, i);
            }
        }
        return sum;
    }
}
